src/main/java/com/metamx/emitter/service/CloudServiceEmitter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2012 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.emitter.service;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.metamx.emitter.core.Emitter;
+
+public class CloudServiceEmitter extends ServiceEmitter
```

Comment:
The class guarantees the use of new fields (env/location/zone). I've considered of adding these fields to the ServiceEmitter but there are several nuances in this approach: the absence of the requirement for new fields and null values in getters.